src/python/weblyzard_api/client/ontogene/__init__.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/python
+# -*- coding: utf8 -*-
+'''
+.. moduleauthor:: Adrian Brasoveanu <adrian.brasoveanu@htwchur.ch>
+'''
+import logging
+import requests
+
+from eWRT.ws.rest import MultiRESTClient
+
+from weblyzard_api.client import (WEBLYZARD_API_URL, WEBLYZARD_API_USER,
+                                  WEBLYZARD_API_PASS, OGER_API_URL)
+from weblyzard_api.xml_content import XMLContent
+
+INTERNAL_PROFILE_PREFIX = 'extras.'
+LOGGER = logging.getLogger('weblyzard_api.client.ontogene.oger')
+
+DEFAULT_MAX_RETRY_DELAY = 15
+DEFAULT_MAX_RETRY_ATTEMPTS = 5
+DAYS_BACK_DEFAULT = 20
+
+#GET URL/{pubmed|pmd}/{txt|bioc|pxml|nxml|pxml.gz}/DOC_ID
+
+class SpecialGetRequest(object):
+    ''' Make a post request and return the connection without
+    reading the data. Allows for finer handling of error codes
+    '''
+    def __init__(self, url, data):
+        self.url = url
+        #self.data = json.dumps({"hashes": data})
+        #self.headers = [{"Content-Type": "application/json"}]
+    
+    def request(self):
+        handler = urllib2.HTTPHandler()
+        opener = urllib2.build_opener(handler)
+        req = urllib2.Request(url=self.url)
+        req.add_header("Content-Type", "application/json")
+        req.get_method = lambda: "GET"
+        req.add_data(self.data)
+        try:
+            conn = opener.open(req)
+        except urllib2.HTTPError as e:
+            conn = e
+        except urllib2.URLError as e:
+            logger.error("Connection refused.. %s", e)
+            raise e
+        return conn
+
+
+class Oger(MultiRESTClient):
+    '''
+    Provides access to the OntoGene Entity Recognition.
+    
+    Currently we support the following types of requests: status, fetch and upload.
+
+    Requests to root are currently not supported since they would simply return an HTML page.
+    
+    A fetch request will retrieve an existing document from a known source:   
+        GET <BASE_URL>/fetch/<SOURCE>/<OUTPUT_FORMAT>/<DOC_ID>
+        POST <BASE_URL>/fetch/<SOURCE>/<OUTPUT_FORMAT>/<DOC_ID>
+    
+    An upload request is used to submit a text to be annotated.
+        POST <BASE_URL>/upload/<INPUT_FORMAT>/<OUTPUT_FORMAT> [/<DOC_ID>]
+    
+    Accepted values:
+        SOURCE: pubmed, pmc
+        INPUT_FORMAT: txt, bioc, pxml, nxml, pxml.gz
+        OUTPUT_FORMAT: bioc, odin, odin_custom, tsv, xml
+
+    '''
+    def __init__(self, url=OGER_API_URL, default_timeout=None):
+        '''
+        :param url: URL of the jeremia web service
+        #:param usr: optional user name
+        #:param pwd: optional password
+        '''
+        #no usr=WEBLYZARD_API_USER, pwd=WEBLYZARD_API_PASS,
+        ##user=usr, password=pwd,
+        MultiRESTClient.__init__(self, service_urls=url, default_timeout=default_timeout)
+        self.profile_cache = []
```

Comment:
what is this for? If not used, please remove

Reply:
removed